Library/Formula/pugixml.rb

```diff
@@ -0,0 +1,58 @@
+class Pugixml < Formula
+  desc "Pugixml is a light-weight C++ XML processing library"
+  homepage "http://pugixml.org"
+  url "http://github.com/zeux/pugixml/releases/download/v1.6/pugixml-1.6.tar.gz"
+  sha256 "473705c496d45ee6a74f73622b175dfb5dde0de372c4dc61a5acb964516cd9de"
+
+  option "with-shared", "Build shared instead of static library"
+
+  depends_on "cmake" => :build
+
+  def install
+    shared = (build.with? "shared") ? "ON" : "OFF"
+
+    args = std_cmake_args
+    args << "-DBUILD_SHARED_LIBS=#{shared}"
+
+    Dir.chdir("scripts")
+
+    system "cmake", ".", *args
+    system "make", "install"
+
+    (lib/"pkgconfig/pugixml.pc").write <<-EOS.undent
```

Comment:
Would prefer to make this a `def` block rather than inline in `def install`, similar to how we handle [Lua](https://github.com/Homebrew/homebrew/blob/master/Library/Formula/lua.rb#L107-L128).